scipy/optimize/_prima/meson.build

```diff
@@ -0,0 +1,101 @@
+libquadmath_dep = cc.find_library('quadmath', required: false)
```

Comment:
There was some discussion about this earlier: https://github.com/scipy/scipy/pull/20964#issuecomment-2171497257

Basically, on Windows the compilation was failing because it complained about not finding the libquadmath symbols, which led to calling out the dependency, but on macOS it complained that it couldn't find libquadmath, but compiled fine without it, hence setting required to false in order to make both Windows and macOS happy. As you can see from the discussion I linked numpy solved this issue [similarly](https://github.com/numpy/numpy/blob/main/numpy/f2py/_backends/meson.build.template#L33)

I will explore the code in that PR you linked and report back.

Reply:
It’s not obvious to me how to apply this code to PRIMA. A search for ‘quadmath’ in that repo leads to 0 results, either in code or issues or PRs. Could this be a meson issue, or as you mentioned in the licensing discussion you linked, a gfortran issue? In any case, I don’t think it’s worth spending a lot of cycles on this given the next issue, unless there’s a license issue here that needs to be addressed.